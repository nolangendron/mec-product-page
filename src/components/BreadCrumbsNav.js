import React from 'react'
import { Breadcrumbs } from '@material-ui/core'
import { Link } from 'react-router-dom'

export const BreadCrumbsNav = () => {
  return (
    <Breadcrumbs style={{ 'margin-left': '1em' }}>
      <Link to="/">
        <h4>Home</h4>
      </Link>
      <h4>Products</h4>
    </Breadcrumbs>
  )
}
