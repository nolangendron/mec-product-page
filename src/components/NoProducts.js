import React from 'react'

export const NoProducts = ({ currentSearch }) => {
  return (
    <div>
      <p>
        We have thousands of products, just not one matching {currentSearch}.
        Try another search!
      </p>
    </div>
  )
}
