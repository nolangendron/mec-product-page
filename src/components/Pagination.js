import React from 'react'
import styled from '@emotion/styled'
import { makeStyles } from '@material-ui/core/styles'
import { ArrowForwardIos, ArrowBackIos } from '@material-ui/icons'

const PaginationStyles = styled('nav')`
  display: flex;
  flex-direction: row;
  justify-content: center;

  .page-button {
    margin: 5px;
    padding: 10px;
    border: none;
    :hover {
      cursor: pointer;
      background: grey;
    }
  }
`

const useStyles = makeStyles({
  root: {
    margin: '5px',
    padding: '10px',
    background: 'lightgrey',
    '&:hover': {
      cursor: 'pointer',
      background: 'grey',
    },
  },
})
export const Pagination = ({
  productPerPage,
  totalProducts,
  paginate,
  paginateArrow,
}) => {
  const pageNumbers = []
  const classes = useStyles()

  for (let i = 1; i <= Math.ceil(totalProducts / productPerPage); i++) {
    pageNumbers.push(i)
  }
  return (
    <PaginationStyles>
      <ArrowBackIos
        onClick={() => paginateArrow('back')}
        className={classes.root}
        fontSize="large"
      />
      {pageNumbers.map((number) => (
        <button
          key={number}
          onClick={() => paginate(number)}
          className="page-button"
        >
          {number}
        </button>
      ))}
      <ArrowForwardIos
        onClick={() => paginateArrow('forward')}
        className={classes.root}
        fontSize="large"
      />
    </PaginationStyles>
  )
}
