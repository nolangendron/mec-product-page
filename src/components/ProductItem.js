import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Card from '@material-ui/core/Card'
import CardActionArea from '@material-ui/core/CardActionArea'
import CardMedia from '@material-ui/core/CardMedia'
import CardContent from '@material-ui/core/CardContent'

const useStyles = makeStyles({
  root: {
    width: 'calc(25% - 2em)',
    minWidth: 300,
    height: '25%',
    margin: '1em',
  },
  media: {
    height: 300,
    width: 300,
    margin: 'auto',
  },
})

export const ProductItem = ({ name, img }) => {
  const classes = useStyles()

  return (
    <Card className={classes.root}>
      <CardActionArea>
        <CardMedia className={classes.media} image={img} title={name} />
        <CardContent>
          <h3>{name}</h3>
        </CardContent>
      </CardActionArea>
    </Card>
  )
}
