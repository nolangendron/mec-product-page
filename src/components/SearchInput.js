import React from 'react'
import PropTypes from 'prop-types'
import styled from '@emotion/styled'
import { Search, Clear } from '@material-ui/icons'
import { colors } from '../styles/colors'

const SearchInputStyles = styled('div')`
  display: flex;
  flex-direction: row;
  margin: 1em;
  min-width: 200px;
  justify-content: ${(props) => (props.landingPage ? 'center' : 'left')};

  input[type='text'] {
    -webkit-appearance: none;
    background-color: ${colors.backgroundMedium};
    color: ${colors.fontLight};
    height: 46px;
    min-height: 40px;
    width: 30%;
    min-width: 190px;
    padding: 10px;
    border: none;
    font-size: 1em;
    :focus {
      color: ${colors.fontLight};
    }
    ::placeholder {
      color: ${colors.fontLight};
      opacity: 1;
    }
  }

  .cancel::-webkit-search-cancel-button {
    -webkit-appearance: none;
  }

  .clear-icon {
    visibility: ${(props) => (props.searchQuery > 0 ? 'visible' : 'hidden')};
  }

  .icon-button {
    height: 46px;
    width: 40px;
    border: none;
    background-color: ${colors.backgroundMedium};
    :hover {
      cursor: pointer;
    }
  }
`

const propTypes = {
  searchQuery: PropTypes.string,
  handleQueryChange: PropTypes.func,
  handleSearch: PropTypes.func,
  landingPage: PropTypes.bool,
  clearSearch: PropTypes.func,
}

export const SearchInput = ({
  searchQuery,
  handleQueryChange,
  handleSearch,
  landingPage,
  clearSearch,
}) => {
  return (
    <SearchInputStyles
      searchQuery={searchQuery && searchQuery.length}
      landingPage={landingPage}
    >
      <form autoComplete="off" onSubmit={handleSearch}>
        <input
          id="search"
          name="search"
          type="text"
          aria-label="Search"
          onChange={handleQueryChange}
          value={searchQuery}
          placeholder="I'm looking for"
          className="cancel"
        />
      </form>
      <button type="reset" onClick={clearSearch} className="icon-button">
        <Clear className="clear-icon" style={{ color: 'white' }} />
      </button>
      <button type="submit" onClick={handleSearch} className="icon-button">
        <Search style={{ color: 'white' }} />
      </button>
    </SearchInputStyles>
  )
}

SearchInput.propTypes = propTypes
