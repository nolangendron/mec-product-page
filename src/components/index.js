export { Header } from './Header'
export { Pagination } from './Pagination'
export { ProductItem } from './ProductItem'
export { ProductsList } from './ProductsList'
export { SearchDetails } from './SearchDetails'
export { BreadCrumbsNav } from './BreadCrumbsNav'
export { SearchInput } from './SearchInput'
export { NoProducts } from './NoProducts'
