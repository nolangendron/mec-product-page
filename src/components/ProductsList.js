import React from 'react'
import styled from '@emotion/styled'
import { ProductItem } from './ProductItem'

const Section = styled('section')`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  width: 100%;

  @media screen and (min-width: 480px) {
    justify-content: center;
  }
`
export const ProductsList = ({ products }) => {
  return (
    <Section>
      {products.map((product, index) => {
        const { name, default_image_urls } = product
        const { main_image_url } = default_image_urls
        return <ProductItem key={index} name={name} img={main_image_url} />
      })}
    </Section>
  )
}
