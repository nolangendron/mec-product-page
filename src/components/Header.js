import React from 'react'
import styled from '@emotion/styled'
import logo from '../images/mec-logo.svg.png'
import SearchInput from './SearchInput'
import { Link } from 'react-router-dom'

const HeaderStyles = styled('header')`
  display: flex;
  flex-direction: row;
  background-color: #191919;
  height: 80px;
  width: 100%;
  align-items: center;

  .flex-item {
    height: 50px;
    margin: 1em;
  }

  .logo {
    width: 50px;
    :hover {
      cursor: pointer;
    }
  }
`
export const Header = ({ searchQuery, handleQueryChange, handleSearch }) => {
  return (
    <HeaderStyles>
      <Link to="/">
        <img className="flex-item logo" src={logo} alt="MEC logo" />
      </Link>
      <SearchInput
        searchQuery={searchQuery}
        handleQueryChange={handleQueryChange}
        handleSearch={handleSearch}
      />
    </HeaderStyles>
  )
}
