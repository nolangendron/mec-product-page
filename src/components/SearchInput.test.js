import React from "react";
import { render } from "@testing-library/react";
import { SearchInput } from "../components/SearchInput";

it("should take a snapshot of SearchInput", () => {
  const { asFragment } = render(<SearchInput />);
  expect(asFragment(<SearchInput />)).toMatchSnapshot();
});
