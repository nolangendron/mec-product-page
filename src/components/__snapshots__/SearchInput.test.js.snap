// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot of SearchInput 1`] = `
<DocumentFragment>
  <div
    class="css-0"
  >
    <form>
      <input
        placeholder="Search for product"
        value=""
      />
    </form>
  </div>
</DocumentFragment>
`;
